src/identity-providers/add/DescriptorSettings.tsx

```diff
@@ -0,0 +1,428 @@
+  const [
+    samlSignatureKeyNameDropdownOpen,
+    setSamlSignatureKeyNameDropdownOpen,
+  ] = useState(false);
+
+  const wantAuthnSigned = useWatch({
+    control: control,
+    name: "config.wantAuthnRequestsSigned",
+  });
+
+  const validateSignature = useWatch({
+    control: control,
+    name: "config.validateSignature",
+  });
+
+  return (
+    <div className="pf-c-form pf-m-horizontal">
+      <FormGroup
+        label={t("ssoServiceUrl")}
+        labelIcon={
+          <HelpItem
+            helpText={th("ssoServiceUrl")}
+            forLabel={t("ssoServiceUrl")}
+            forID="kc-sso-service-url"
+          />
+        }
+        fieldId="kc-sso-service-url"
+        isRequired
+        validated={
+          errors.config?.authorizationUrl
+            ? ValidatedOptions.error
+            : ValidatedOptions.default
+        }
+        helperTextInvalid={t("common:required")}
+      >
+        <TextInput
+          type="text"
+          data-testid="sso-service-url"
+          id="kc-sso-service-url"
+          name="config.singleSignOnServiceUrl"
+          ref={register({ required: true })}
+          validated={
+            errors.config?.singleSignOnServiceUrl
+              ? ValidatedOptions.error
+              : ValidatedOptions.default
+          }
+          isReadOnly={readOnly}
+        />
+      </FormGroup>
+
+      <FormGroup
+        label={t("singleLogoutServiceUrl")}
+        labelIcon={
+          <HelpItem
+            helpText={th("singleLogoutServiceUrl")}
+            forLabel={t("singleLogoutServiceUrl")}
+            forID="single-logout-service-url"
+          />
+        }
+        fieldId="single-logout-service-url"
+        validated={
+          errors.config?.singleLogoutServiceUrl
+            ? ValidatedOptions.error
+            : ValidatedOptions.default
+        }
+        helperTextInvalid={t("common:required")}
+      >
+        <TextInput
+          type="text"
+          id="single-logout-service-url"
+          name="config.singleLogoutServiceUrl"
+          isReadOnly={readOnly}
+        />
+      </FormGroup>
+
+      <SwitchField
+        field="config.backchannelSupported"
+        label="backchannelLogout"
+        isReadOnly={readOnly}
+      />
+
+      <FormGroup
+        label={t("nameIdPolicyFormat")}
+        labelIcon={
+          <HelpItem
+            helpText={th("nameIdPolicyFormat")}
+            forLabel={t("nameIdPolicyFormat")}
+            forID="kc-nameIdPolicyFormat"
+          />
+        }
+        fieldId="kc-nameIdPolicyFormat"
+        helperTextInvalid={t("common:required")}
+      >
+        <Controller
+          name="config.nameIDPolicyFormat"
+          defaultValue={t("persistent")}
+          control={control}
+          render={({ onChange, value }) => (
+            <Select
+              toggleId="kc-nameIdPolicyFormat"
+              onToggle={() =>
+                setNamedPolicyDropdownOpen(!namedPolicyDropdownOpen)
+              }
```

Comment:
There is no need to manage this state ourselves as the callback provides an argument for it.
```suggestion
              onToggle={(isExpanded) => setNamedPolicyDropdownOpen(isExpanded)}
```